### PR TITLE
SONARJAVA-2154 Remove usages of Multisets and Multimaps from java-checks module

### PR DESCRIPTION
**Pull request by margarita-nedzelska-sonarsource**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java [hunk 1]
```diff
@@ -53,22 +50,22 @@
...
-      .put("removal", "java:S5738")
-      .put("serial", "java:S2057")
-      .put("static", "java:S2209")
-      .put("rawtypes", "java:S3740")
+  private static final Map<String, Set<String>> JAVAC_WARNING_SUPPRESSING_RULES = MapBuilder.<String, Set<String>>newMap()
+      .put("cast", Collections.singleton("java:S1905"))
+      .put("deprecation", Collections.singleton("java:S1874"))
+      .put("dep-ann", Collections.singleton("java:S1123"))
+      .put("divzero", Collections.singleton("java:S3518"))
+      .put("empty", ImmutableSet.of("java:S1116", "java:S108"))
```
Comment (margarita-nedzelska-sonarsource) [inline java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java:58]: Will remove it after PR with ImmutableSet is merged

Comment (margarita-nedzelska-sonarsource) [inline java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java:58]:
```suggestion
      .put("empty", SetUtils.immutableSetOf("java:S1116", "java:S108"))
```

Comment (m-g-sonar) [reply java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java:58]: go go :)

File: java-frontend/src/main/java/org/sonar/java/collections/SetUtils.java [hunk 2]
```diff
@@ -23,6 +23,8 @@
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
```
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/collections/SetUtils.java:27]: unused import. Can you fix it?

Comment (margarita-nedzelska-sonarsource) [reply java-frontend/src/main/java/org/sonar/java/collections/SetUtils.java]: fixed

**Comment by m-g-sonar** [comment]
LGTM!

**Pull request merged**